### DISCUSSION TITLE
circular import error for rdkit built from source code

### DISCUSSION THREAD

**Discussion by zengkaipeng** [rdkit/rdkit#8033] [Q&A] [answered]
I encountered some issues while building RDKit version release_2024_09, following the instructions from [https://github.com/rdkit/rdkit/blob/Release_2024_09/Docs/Book/Install.md](https://github.com/rdkit/rdkit/blob/Release_2024_09/Docs/Book/Install.md). When I run `ctest`, I found the following errors reported.

![1732455320261](https://github.com/user-attachments/assets/2de1ce81-7188-49d2-bdfa-c902adf50a08)

During the compilation process, a new `rdkit` folder was created under the `build` directory, and the compiled `so` files were placed inside it. Is this correct? What is the difference between this `rdkit` and the `rdkit` folder shown in the project repo? It seems that the compiled `so` files should be placed under the `rdkit` folder in the repo, not in the `rdkit` subfolder under the `build` directory. Should I modify the values of the `RDK_INSTALL` variables in the following command to complete the correct compilation?

```bash
cmake -DPy_ENABLE_SHARED=1 \
  -DRDK_INSTALL_INTREE=ON \
  -DRDK_INSTALL_STATIC_LIBS=OFF \
  -DRDK_BUILD_CPP_TESTS=ON \
  -DPYTHON_NUMPY_INCLUDE_PATH="$(python -c 'import numpy ; print(numpy.get_include())')" \
  -DBOOST_ROOT="$CONDA_PREFIX" \
  ..
```

**Comment by bp-kelley** [comment]
This looks like you missed:

`make install`

**Reply by bp-kelley** [reply, marked as answer]
The .so should be copied into the existing rdkit directory with the existing python files

**Reply by zengkaipeng** [reply]
Thank you~!
